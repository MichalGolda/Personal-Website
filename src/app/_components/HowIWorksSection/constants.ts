export enum PhaseCardColors {
  PHASE_FIRST = "before:bg-[#30E6FF] text-[#30E6FF]",
  PHASE_SECOND = "before:bg-[#FFB930] text-[#FFB930]",
  PHASE_THIRD = "before:bg-[#FF4930] text-[#FF4930]",
  PHASE_FOURTH = "before:bg-[#B130FF] text-[#B130FF]",
}
