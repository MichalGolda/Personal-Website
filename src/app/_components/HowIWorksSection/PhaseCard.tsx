import { PhaseCardColors } from "./constants";

export type PhaseCardProps = {
  title: string;
  headline: string;
  body: string;
  color: PhaseCardColors;
  className?: string;
};

const PhaseCard: React.FC<PhaseCardProps> = ({
  title,
  headline,
  color,
  body,
  className,
  ...props
}) => {
  return (
    <div
      className={`flex flex-col gap-y-4 border border-solid border-lightGrey p-4 bg-white ${className}`}
      {...props}
    >
      <div className="flex flex-col">
        <span
          className={`before:block flex items-center gap-x-2 text-sm md:text-base font-semibold uppercase before:w-2 before:h-2 ${color}`}
        >
          {headline}
        </span>
        <h4 className="text-lg md:text-xl lg:text-2xl text-secondary font-bold text-secondary">
          {title}
        </h4>
      </div>
      <p className="text-body text-sm md:text-base">{body}</p>
    </div>
  );
};

export default PhaseCard;
