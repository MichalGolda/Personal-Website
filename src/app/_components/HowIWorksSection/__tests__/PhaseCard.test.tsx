import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import PhaseCard from "../PhaseCard";
import { PhaseCardColors } from "../constants";

describe("PhaseCard component", () => {
  it("renders without errors", () => {
    const { container } = render(
      <PhaseCard
        title="test title"
        color={PhaseCardColors.PHASE_FIRST}
        headline="test headline"
        body="test body"
        className="test class"
      />
    );

    const title = screen.getByText("test title");
    const headline = screen.getByText("test headline");

    expect(container.firstChild).toHaveClass("test class");

    expect(title).toBeInTheDocument();
    expect(title.tagName).toBe("H4");

    expect(headline).toBeInTheDocument();
    expect(headline).toHaveClass("before:bg-[#30E6FF]");
    expect(headline).toHaveClass("text-[#30E6FF]");
    expect(headline.tagName).toBe("SPAN");
  });

  it("renders unchanged", () => {
    const { container } = render(
      <PhaseCard
        title="test title"
        color={PhaseCardColors.PHASE_FIRST}
        headline="test headline"
        body="test body"
        className="test class"
      />
    );

    expect(container).toMatchSnapshot();
  });
});
