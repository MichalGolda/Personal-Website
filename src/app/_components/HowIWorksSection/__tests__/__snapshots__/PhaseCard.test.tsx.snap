// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhaseCard component renders unchanged 1`] = `
<div>
  <div
    class="flex flex-col gap-y-4 border border-solid border-lightGrey p-4 bg-white test class"
  >
    <div
      class="flex flex-col"
    >
      <span
        class="before:block flex items-center gap-x-2 text-sm md:text-base font-semibold uppercase before:w-2 before:h-2 before:bg-[#30E6FF] text-[#30E6FF]"
      >
        test headline
      </span>
      <h4
        class="text-lg md:text-xl lg:text-2xl text-secondary font-bold text-secondary"
      >
        test title
      </h4>
    </div>
    <p
      class="text-body text-sm md:text-base"
    >
      test body
    </p>
  </div>
</div>
`;
